controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java

```diff
@@ -163,6 +163,7 @@ public Slime toSlime(Application application) {
         root.setDouble(queryQualityField, application.metrics().queryServiceQuality());
         root.setDouble(writeQualityField, application.metrics().writeServiceQuality());
         application.pemDeployKey().ifPresent(pemDeployKey -> root.setString(pemDeployKeyField, pemDeployKey));
+        application.legacyRotation().ifPresent(rotation -> root.setString(deprecatedRotationField, rotation.asString()));
```

Comment:
The `legacyRotation` field isn't really needed as we could've just written `deprecatedRotationField` directly from `application.rotations()` here. This is fine for now though, but let's remove it in the next version.

Reply:
Yup. More important to have the issue fixed :) Removing it with the legacy (de)serialisation should be OK once this is deployed.

Reply:
I agree, but I didn't feel brave enough to make assumptions about order or anything like that :)